### PR TITLE
fix some deprecated qt declarations and compiler warnings

### PR DESCRIPTION
**Pull request by oberon2007**
Most of the adjustments are just backported from master branch.
Like this, branch 2.2 now builds without warnings with **qt5.8** and **gcc 6.3.1** on Manjarolinux, except for one `-Wunused-parameter`:
```
MuseScore/libmscore/durationtype.cpp: In constructor ‘Ms::TDuration::TDuration(const Ms::Fraction&, bool, int, Ms::TDuration::DurationType)’:
MuseScore/libmscore/durationtype.cpp:417:46: warning: unused parameter ‘truncate’ [-Wunused-parameter]
 TDuration::TDuration(const Fraction& l, bool truncate, int maxDots, DurationType maxType)
```

### CODE REVIEW

File: libmscore/accidental.cpp [hunk 1]
```diff
@@ -367,7 +367,7 @@ AccidentalType Accidental::value2subtype(AccidentalVal v)
             case AccidentalVal::FLAT:    return AccidentalType::FLAT;
             case AccidentalVal::FLAT2:   return AccidentalType::FLAT2;
             default:
-                  qFatal("value2subtype: illegal accidental val %hhd", v);
+                  qFatal("value2subtype: illegal accidental val %d", static_cast<int>(v));
```
Comment (Jojo-Schmitz) [inline libmscore/accidental.cpp]: Why not simply `int(v)` here too?

File: libmscore/harmony.cpp [hunk 2]
```diff
@@ -1346,7 +1346,7 @@ void Harmony::render(const QList<RenderAction>& renderList, qreal& x, qreal& y,
                         }
                   }
             else
-                  qDebug("========unknown render action %hhd", a.type);
+                  qDebug("Harmony::render(): unknown render action %d", static_cast<int>(a.type));
```
Comment (Jojo-Schmitz) [inline libmscore/harmony.cpp]: That Harmony::render gets added automatically by qDebug, so is redundant here

File: thirdparty/ofqf/qoscserver.cpp [hunk 3]
```diff
@@ -82,8 +82,8 @@ void QOscServer::readyRead()
         		      for ( ; i<size && data[ i ] != char( 0 ); ++i )
         		            path += data[ i ];
         
-        			    while ( data[ i ] != ',' ) ++i;
-        			    ++i;
+        			  while ( data[ i ] != ',' ) ++i;
+        			  ++i;
         		      while ( data[ i ] != char( 0 ) )
```
Comment (Jojo-Schmitz) [inline thirdparty/ofqf/qoscserver.cpp]: We indent 6 spaces. And probably don't touch 3rd party stuff for this.

File: mstyle/mstyle.cpp [hunk 4]
```diff
@@ -1290,8 +1290,8 @@ bool MgStyle::drawFrameGroupBoxPrimitive(const QStyleOption* option, QPainter* p
             return true;
 
       // no frame for flat groupboxes
-      QStyleOptionFrameV2 fOpt2(*fOpt);
-      if (fOpt2.features & QStyleOptionFrameV2::Flat)
+      QStyleOptionFrame fOpt2(*fOpt);
+      if (fOpt2.features & QStyleOptionFrame::Flat)
```
Comment (Jojo-Schmitz) [inline mstyle/mstyle.cpp]: Sure this works with Qt 5.4?

**Comment by oberon2007** [comment]
adjusted.

File: mstyle/mstyle.cpp (hunk 4 above)
Comment (oberon2007) [inline mstyle/mstyle.cpp]: Seems it was only introduced in 5.5
http://doc.qt.io/qt-5/qframe.html#initStyleOption

Comment (Jojo-Schmitz) [inline mstyle/mstyle.cpp]: So we can't change it for MuseScore 2.x, which requires Qt 5.4

**Comment by Jojo-Schmitz** [comment]
2.x requires Qt 5.4

**Comment by Jojo-Schmitz** [comment]
And why not fixing the Wunused warning?

**Comment by Jojo-Schmitz** [comment]
There is more of these int casts to simplify...

**Comment by oberon2007** [comment]
How do you fix the Wunused warning? 😉 I don't know I'm afraid...

**Comment by Jojo-Schmitz** [comment]
`bool /*truncate*/`

**Comment by oberon2007** [comment]
Ok. sure. 😆

File: thirdparty/ofqf/qoscserver.cpp (hunk 3 above)
Comment (oberon2007) [inline thirdparty/ofqf/qoscserver.cpp]: Yeah, indenting of ofqf is a mess. Just this one looks like just a mistake and gcc says:
```
/home/bernhard/local-build/musescore-2.2/src/MuseScore/thirdparty/ofqf/qoscserver.cpp: In member function ‘void QOscServer::readyRead()’:
MuseScore/thirdparty/ofqf/qoscserver.cpp:82:17: warning: this ‘for’ clause does not guard... [-Wmisleading-indentation]
                 for ( ; i<size && data[ i ] != char( 0 ); ++i )
                 ^~~
MuseScore/thirdparty/ofqf/qoscserver.cpp:85:16: note: ...this statement, but the latter is misleadingly indented as if it is guarded by the ‘for’
                while ( data[ i ] != ',' ) ++i;
```

Comment (Jojo-Schmitz) [inline thirdparty/ofqf/qoscserver.cpp]: But replacing indent by 6 with indent by 4 ain't gonna fix it...

File: mstyle/mstyle.cpp (hunk 4 above)
Comment (oberon2007) [inline mstyle/mstyle.cpp]: I'm still not sure about QStyleOptionFrame. The comment in the link I posted I think only refers to `initStyleOption`, not `QStyleOptionFrame`
Unfortunately I cannot check older qt versions here, but what I found is that qt542 contains all three QStyleOptionFrame QStyleOptionFrameV2 and QStyleOptionFrameV3.
That travis was able to build the commit does not tell us anything? Will it only be a runtime problem?
Would you in any case just prefer to keep the old compat declarations?

File: thirdparty/ofqf/qoscserver.cpp (hunk 3 above)
Comment (oberon2007) [inline thirdparty/ofqf/qoscserver.cpp]: I didn't replace by 4. Have a closer look. There are also Tabs involved in this file. The commit fixes the warning, indeed.

File: mstyle/mstyle.cpp (hunk 4 above)
Comment (Jojo-Schmitz) [inline mstyle/mstyle.cpp]: Dunno, would need to check, so far I took your word that it was added in5.5

File: thirdparty/ofqf/qoscserver.cpp (hunk 3 above)
Comment (Jojo-Schmitz) [inline thirdparty/ofqf/qoscserver.cpp]: Hmm, ok, can't see this from here.

File: mstyle/mstyle.cpp (hunk 4 above)
Comment (Jojo-Schmitz) [inline mstyle/mstyle.cpp]: But if Travis builds it, it should be ok

**Comment by lasconic** [comment]
Hi, thank you for your contribution. I'm testing it on macOS as we speak. 
In order to merge it, I need you to sign the MuseScore CLA https://musescore.org/en/cla. Can you do that and give me your MuseScore username? Thanks!

**Comment by lasconic** [comment]
It works with 5.4.2 on macOS.

**Comment by oberon2007** [comment]
You're welcome! My username is the same as the github username oberon2007. CLI signed just now :)

File: libmscore/durationtype.cpp [hunk 5]
```diff
@@ -414,7 +414,7 @@ Fraction TDuration::fraction() const
       }
 
 // Longest TDuration that fits into Fraction. Must fit exactly if truncate = false.
-TDuration::TDuration(const Fraction& l, bool truncate, int maxDots, DurationType maxType)
+TDuration::TDuration(const Fraction& l, bool /*truncate*/, int maxDots, DurationType maxType)
```
Comment (Jojo-Schmitz) [inline libmscore/durationtype.cpp]: Actually truncate is used in the QAssert, so a QUnused might be the better choice, maybe combined with NDEBUG

**Comment by oberon2007** [comment]
Should be ready. For me it builds without any warning now.

File: CMakeLists.txt [hunk 6]
```diff
@@ -163,8 +163,8 @@ else (APPLE)
       set(CMAKE_CXX_FLAGS_RELEASE "-std=c++11 -mno-ms-bitfields -O2 -DNDEBUG -DQT_NO_DEBUG")
       set(CMAKE_EXE_LINKER_FLAGS "-Wl,--large-address-aware")
    else (MINGW)
-      set(CMAKE_CXX_FLAGS_DEBUG   "-std=c++11 -fPIC -g")
-      set(CMAKE_CXX_FLAGS_RELEASE "-std=c++11 -fPIC -O2 -DNDEBUG -DQT_NO_DEBUG")
+      set(CMAKE_CXX_FLAGS_DEBUG   "-std=gnu++11 -fPIC -g")
+      set(CMAKE_CXX_FLAGS_RELEASE "-std=gnu++11 -fPIC -O2 -DNDEBUG -DQT_NO_DEBUG")
```
Comment (Jojo-Schmitz) [inline CMakeLists.txt]: I don't think we should be using gnu++11

Comment (oberon2007) [inline CMakeLists.txt]: Well, it was intended for qt from 5.7, you're right. How about adding ``-fno-inline`` anyway?

Comment (oberon2007) [inline CMakeLists.txt]: Ok, forget that. It seems using ``c++11`` again also takes care of ``__NO_INLINE__`` 😉

Comment (oberon2007) [inline CMakeLists.txt]: Although, now I get ``not used because `__STRICT_ANSI__' not defined`` :laugh: beh

**Comment by oberon2007** [comment]
I didn't get any complaint from the compiler about Q_UNUSED with ``CMAKE_BUILD_TYPE=DEBUG`` but added ``#ifdef NDEBUG`` now anyway - I guess it's correct this way.
Now the question is still about ``CMAKE_CXX_FLAGS`` for non-APPLE/non-MINGW.
For me building on an Archlinux-based more uptodate system what I need is ``-std=gnu++11`` or I will see tons of pch errors and build takes forever like that. For the debug build ``-fno-inline`` is needed additionally.
I have committed this now in any case to see if travis has any issues building it like that.
In the end it's for you to decide. I think it makes sense, since Windows and Apple systems will not be affected and Linux systems tend to run newer compilers and qt-environments...

**Comment by oberon2007** [comment]
So  apparenty travis, using **gcc-4.9** and **qt5.4.2** is also happy.

**Comment by oberon2007** [comment]
Of course if you prefer not to adopt the compiler flags in you **CMakeLists** we can add them in the Archlinux and AUR PKGBUILDs instead.

**Comment by lasconic** [comment]
Ok looks good. Some commits are not really useful as independent ones. Could you squash, possibly in a single commit? http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

**Comment by oberon2007** [comment]
Uhm. The last commit is the squashed one, but the others are still showing in the history. Don't know if I did something wrong or if that's because the commits were already pushed to remote.
Should I just apply the squashed commits to a new branch and open a new PR?

**Comment by lasconic** [comment]
Sure it works too.

**Comment by oberon2007** [comment]
squashed into https://github.com/musescore/MuseScore/pull/3187

**Comment by Jojo-Schmitz** [comment]
one way is `git reset <sha-before your first commit>`, followed by a `git commit -m"your commit message"`

**Comment by oberon2007** [comment]
I'll try that next time :)
The problem with the rebase like described in the link is that it will not let you push to remote without doing another pull, so I had to merge ... I think it will only work like that locally before you have pushed the commits, maybe?

**Comment by Jojo-Schmitz** [comment]
Maybe try a `git pull --rebase upstream 2.2`

**Comment by oberon2007** [comment]
That again leads to
```
$ git status
On branch 2.2
Your branch and 'origin/2.2' have diverged,
and have 26 and 14 different commits each, respectively.
  (use "git pull" to merge the remote branch into yours)
nothing to commit, working tree clean
```

**Comment by oberon2007** [comment]
Well the new PR is fine now anyway, but for the sake of learning more about git I just also tried what you suggested first:
```
git reset 20cc6826d0e9a97dbfe6824489055095a32d9f56
git add .
git commit -m"all the good stuff"
git push
```
So far so good, but now:
```
 ! [rejected]            2.2 -> 2.2 (non-fast-forward)
error: failed to push some refs to 'https://github.com/oberon2007/MuseScore.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
```
I could still reset --hard and then use the patch from from the other new branch to re-apply the changes, but there must be another clean way, too, no? 😉

**Comment by oberon2007** [comment]
I figured it out now. The missing important bit of information was that at the end of the rebase you need to ``git push --force``
https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request

**Pull request closed without merging**